### ISSUE TITLE
Kpp is wrong

### ISSUE THREAD

**Issue by adamsmd** [kframework/k-legacy#177]
`kpp` strips the comments from a K definition file.  However, as it does not use the same parser as the main basic parser, it is actually wrong.  (e.g., it does string handling wrong.)

**Comment by grosu** [comment]
I personally think that we should maintain the kpp functionality.  Beginners or students typically start their definitions by taking an existing definition and modifying it.  One request which I heard several times in class was that they would even like an option where the line numbers are not maintained, that is, where the multiple whitespace lines in the current output are replaced by just one new line.

What I am not sure about is whether it should be a separate tool or just part of kompile.  I know it may sound strange to be part of kompile, but recall that we eventually want kompile to spit out the resulting definition after any kompilation step.  kpp would just be the definition after the first step.

So, what do you guys think: should we keep kpp as a separate tool, or we should incorporate through options in kompile?

**Comment by dlucanu** [comment]
I agree to have it as a part of kompile, but we have to clearly define 
the compilation steps because now they are used beginners. May be we 
should split them in two categories: basic and advanced.

Dorel

**Comment by adamsmd** [comment]
(1) Do you care about preserving the formatting of the code?

(2) Do you care about doing this on code that doesn't properly parse?

If the answers are both "no", then it will be easy to just add a 
"backend" to kompile that does this.  Otherwise, there are subtleties 
that would require I better understand the use-cases for Kpp.

**Comment by adamsmd** [comment]
I don't understand what you have in mind when you are talking about 
basic versus advanced compilation steps.  Could you elaborate?

Michael

**Comment by grosu** [comment]
Good questions.  To keep it simple, I think it is OK to assume that the answer is NO.  If people want to just strip out the comments blindly then they can write their own scripts, as students in cs422 actually did.

Grigore
